cot/src/form/types.rs

```diff
@@ -0,0 +1,351 @@
+//! Form Field Types for Cot
```

Comment:
still deliberating on whether it's useful to have this file than put everything in `fields.rs`

Reply:
I think it will be useful, but perhaps we could move it directly to `crate::types`, given it's useful not only in forms, but also in auth. I'm not 100% sold on the name `types`, though; it's very generic. I'm thinking about something in the lines of `value_types` or `data`.

@seqre do you have any additional views on this?

Reply:
>I think it will be useful, but perhaps we could move it directly to `crate::types`, given it's useful not only in forms, but also in auth.

I agree, `Password` and `Email` (and other similar types) are so common that it makes sense to extract them to their own place, so they can be reused in different situations.

>I'm not 100% sold on the name `types`, though; it's very generic. I'm thinking about something in the lines of `value_types` or `data`.

I'm bad at naming as well, but some other ideas that come to my mind: `common_types`, `user_types`, or `user_data`. I'd be against plain `data` as it's too generic like `types`.

Reply:
`common_types` sounds good to me, but I also like `value_types` 🤔

Reply:
ended up going with `common_types`. Anyone with an objection, speak now or forever hold your peace 😅